### PR TITLE
Make PromptKit version visible: session banner + CLI update check

### LINKED ISSUE #248: Announce PromptKit version when bootstrap/manifest loads
## Summary

Make the PromptKit version visible to users the moment the composition engine starts, so users can tell which release of PromptKit they are running and so session logs carry provenance that is useful when debugging or reporting issues.

## Motivation

Today, when `bootstrap.md` is loaded — either manually by pointing an LLM at the repo, or through the `promptkit interactive` (`npx`) flow that stages content into a temp dir — the LLM begins composing prompts without identifying which version of PromptKit is actually in play. That makes it hard for users and maintainers to:

- confirm which release is active in a given session,
- correlate bug reports or unexpected template behavior with a specific version,
- tell apart drift between local edits, a pinned `npx` install, and the upstream main branch.

## Proposed change

Add a one-line version banner at the start of every PromptKit session:

- After step 1 of `How to Begin` in `bootstrap.md`, the composition engine reads the top-level `version:` field from `manifest.yaml` and emits `PromptKit v<version> loaded.` before any other output.
- If the field is missing or unreadable, emit `PromptKit (version unknown) loaded.` — do not fabricate a version number.
- Re-announce when `bootstrap.md` or `manifest.yaml` is re-read later in the same session.

Because all three entry-point skills (`/promptkit`, `/bootstrap`, `/boot`) and the `promptkit interactive` CLI all converge on `bootstrap.md`, one edit covers every activation path — manual load, slash command, and npx.

## Tracked work

- [x] Edit `bootstrap.md` step 1 to announce the manifest version (branch: `announce-promptkit-version-on-load`)
- [ ] Open a PR from that branch into `main`
- [ ] (Optional, belt-and-suspenders) Extend `tests/validate-manifest.py` to assert `manifest.yaml.version` matches `cli/package.json.version` so the in-session banner and `promptkit --version` can never drift

## Benefits

- **Provenance**: every session self-identifies its PromptKit version, making bug reports actionable without users having to dig for it.
- **Debugging**: when a template behaves differently than expected, the banner immediately confirms whether the user is on the expected release.
- **Consistency**: manual load and `npx` users get the same signal, since both paths read the same staged `bootstrap.md` + `manifest.yaml`.

## Notes

- No change to `cli/bin/cli.js` is needed for this feature; `promptkit --version` already reports the CLI package version. The banner is the session-side counterpart.
- `manifest.yaml.version` and `cli/package.json` are already kept aligned per `cli/specs/design.md` v0.6.1.

### LINKED ISSUE #250: Notify users when a newer PromptKit CLI version is available
## Summary

The PromptKit CLI (`@alan-jowett/promptkit`) currently has no mechanism to tell users that a newer version is available. Users who installed via `npm i -g` (or via an old `npx` cache) can silently run an outdated build indefinitely. Any ""update banner"" users may have seen in the past belongs to npm itself or to the downstream LLM CLI (`copilot`, `claude`, `codex`), not to PromptKit.

## Motivation

- Users running against stale PromptKit content may hit bugs that are already fixed upstream.
- Triage is harder when there is no in-CLI signal that the user is behind.
- This is a solved problem in the Node ecosystem; PromptKit should follow the same convention (`npm`, `yarn`, `vercel`, `gh` all do it).

## Proposed change

Add a lightweight update check to `cli/bin/cli.js` for the default `interactive` command.

**Behavior**

1. Query `https://registry.npmjs.org/<pkg>/latest` with a short timeout (≈ 1500 ms).
2. Compare `json.version` against `package.json.version` using basic semver major.minor.patch ordering.
3. If `latest > current`, print a one-box banner before spawning the LLM CLI:

   ```
   +----------------------------------------+
   | Update available: 0.6.1 -> 0.7.0       |
   | Run: npm i -g @alan-jowett/promptkit   |
   +----------------------------------------+
   ```

4. Cache the result in `~/.promptkit/update-check.json` for 24 hours so subsequent launches don't hit the network.
5. **Never** block, retry, or fail the CLI on network/cache errors — silently skip and let the CLI continue.

**Opt-out / non-interactive safety**

- Skipped when `NO_UPDATE_NOTIFIER=1`.
- Skipped when `CI` env var is set.
- Skipped when `stdout` is not a TTY (scripted/piped usage).
- Skipped when `--no-update-check` flag is passed.
- Skipped for non-interactive subcommands (`list`, `search`, `show`, `--version`) to keep machine-readable output clean.

**Implementation constraints**

- **No new runtime dependencies.** Use Node's built-in `https` module. Do not pull in `update-notifier`, `got`, `semver`, etc.
- Pure functions (version parsing, comparison, banner formatting) go in a new `cli/lib/update-check.js` module with unit tests that don't touch the network.
- Network fetch timeout is hard-capped; no pathological hang regardless of registry responsiveness.

## Acceptance criteria

- Fresh session on an outdated install prints the banner once, then spawns the LLM CLI normally.
- Second invocation within 24 h hits the cache and still prints the banner — no second network call.
- Network failures, cache I/O failures, and malformed JSON never surface to the user.
- `NO_UPDATE_NOTIFIER=1`, `CI=1`, `--no-update-check`, and non-TTY stdout all suppress the banner.
- `promptkit --version` and `promptkit list --json` output nothing extra.
- Unit tests cover version comparison edge cases (leading `v`, prerelease tags, equal versions, major/minor/patch bumps).

## Out of scope

- Update detection when `bootstrap.md` is loaded manually (no CLI involved). Tracked separately if we pursue it; it requires agent-runtime web tools and is not universally available.

### PR DESCRIPTION
**Pull request by abeltrano**
Closes #248.
Closes #250.

This PR adds two small, complementary pieces of version visibility to PromptKit.

---

## 1. Announce the loaded PromptKit version (#248)

One surgical edit to `bootstrap.md` step 1 of `How to Begin`. After reading `manifest.yaml`, the composition engine now reads the top-level `version:` field and emits a banner before any other output. If the field is missing or unreadable, a `(version unknown)` fallback is emitted — no version is ever fabricated. The engine re-announces whenever `bootstrap.md` or `manifest.yaml` is re-read.

All three entry-point skills (`/promptkit`, `/bootstrap`, `/boot`) and the `promptkit interactive` (`npx`) flow converge on `bootstrap.md`, so this one edit covers every activation path — manual load, slash command, and CLI.

### New session-start output

Against the current repo (`manifest.yaml` has `version: "0.6.1"`), the first line of any new session is:

```
PromptKit v0.6.1 loaded.
```

If `manifest.yaml` ever ships without a readable `version:` field, the engine emits instead:

```
PromptKit (version unknown) loaded.
```

---

## 2. Notify users when a newer CLI version is available (#250)

Adds a best-effort update check to `promptkit interactive`. Follows the standard pattern used by `npm`, `yarn`, `vercel`, and `gh`, but implemented with **no new runtime dependencies** (pure built-in `https`).

### Behavior

- Queries `https://registry.npmjs.org/@alan-jowett/promptkit/latest` with a hard ~1500 ms wall-clock deadline (overall `setTimeout` that destroys the request — not just socket-inactivity).
- Compares `json.version` against `package.json.version` with simple `major.minor.patch` ordering (strips a leading `v`, ignores prerelease/build suffix). Values that don't parse as semver are dropped — never cached, never returned.
- On newer version, prints a boxed banner **before** spawning the LLM CLI:

  ```
  +----------------------------------------+
  | Update available: 0.6.1 -> 0.7.0       |
  | Run: npm i -g @alan-jowett/promptkit   |
  +----------------------------------------+
  ```

- Caches the result in `~/.promptkit/update-check.json` for 24 h; subsequent launches skip the network but still show the banner if a newer version is cached. Cached values are re-validated with `parseVersion` on read, so a poisoned cache entry is treated as a miss and refetched.
- Network, DNS, timeout, HTTP-non-200, malformed-JSON, and cache I/O failures are **all swallowed** — the check is strictly best-effort and can never block or break the CLI.
- Response body is capped at 64 KB to defend against a misbehaving registry.

### Opt-out / non-interactive safety

- `NO_UPDATE_NOTIFIER=1` — suppresses the check.
- `CI` env var set — suppresses.
- `stdout` is not a TTY (piped/scripted usage) — suppresses.
- `--no-update-check` flag — suppresses.
- Non-interactive subcommands (`list`, `search`, `show`, `--version`) never run the check, keeping machine-readable output clean.

### Surface area

- `cli/lib/update-check.js` — new module. Pure functions (`parseVersion`, `isNewer`, `formatBanner`, `suppressionReason`) plus the one network function (`fetchLatest`) behind `checkForUpdate`. `checkForUpdate` dispatches `readCache` / `writeCache` / `fetchLatest` through `_internals` so tests can stub them without filesystem or network I/O.
- `cli/bin/cli.js` — `interactive` action becomes `async`, calls `checkForUpdate` inside a try/catch before `launchInteractive`, gated by the new `--no-update-check` flag.
- `cli/tests/update-check.test.js` — new unit tests covering version parsing/comparison/banner/suppression, plus the cache/TTL branch (fresh hit, expired refetch, pkg-name mismatch, unparseable cached value treated as miss, unparseable fetched value not cached, hard-deadline behavior). No network I/O in tests.
- `cli/tests/cli.test.js` — test harness extended to copy `lib/update-check.js` into its temp CLI root.
- `cli/package.json` — `test` script adds the new test file. No new dependencies.

### Test status

`npm test` in `cli/` passes 72/73. The one remaining failure (`TC-CLI-082/083 gh-copilot spawned with originalCwd and --add-dir for staging dir` in `launch.test.js:335`) **also fails on `main`** and is unrelated to this change — it is a pre-existing flaky/environmental test.

`python tests/validate-manifest.py` — OK.

---

## Version-source consistency

`manifest.yaml.version` (what the in-session banner reads) and `cli/package.json.version` (what `promptkit --version` and the update check read) are already kept aligned per `cli/specs/design.md` v0.6.1, so all three user-visible version surfaces report the same number for any published release. A follow-up to enforce this in `tests/validate-manifest.py` is captured in #248.

## Files changed

- `bootstrap.md` (+10 lines)
- `cli/bin/cli.js` (+18 / -1)
- `cli/lib/update-check.js` (new, 197 lines)
- `cli/package.json` (+1 / -1 — test script)
- `cli/tests/cli.test.js` (+6 lines — harness)
- `cli/tests/update-check.test.js` (new, 329 lines)

No persona, protocol, template, format, or taxonomy content touched.

### CODE REVIEW

File: cli/lib/update-check.js [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+    const done = (value) => {
+      if (settled) return;
+      settled = true;
+      resolve(value);
+    };
+    try {
+      const req = https.get(
+        url,
+        { timeout: FETCH_TIMEOUT_MS, headers: { Accept: "application/json" } },
+        (res) => {
```
[the remark this answers is not in the document]
Comment (abeltrano) [reply cli/lib/update-check.js:88]: Good catch — pushed a fix in 4b3f564. `fetchLatest` now sets an overall `setTimeout(req.destroy, FETCH_TIMEOUT_MS)` (cleared in `done`) so the 1500ms cap is a true wall-clock deadline, not just socket-inactivity.

File: cli/lib/update-check.js [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+    typeof cache.latest === "string" &&
+    typeof cache.checkedAt === "number" &&
+    now - cache.checkedAt < CACHE_TTL_MS
+  ) {
+    latest = cache.latest;
+  } else {
+    latest = await fetchLatest(pkgName);
+    if (latest) {
+      writeCache({ pkg: pkgName, latest, checkedAt: now });
+    }
```
[the remark this answers is not in the document]
Comment (abeltrano) [reply cli/lib/update-check.js:182]: Fixed in 4b3f564. `fetchLatest` now drops anything that doesn't parse as semver, and `checkForUpdate` re-validates the cached `latest` with `parseVersion` before using it — so a pre-existing bad cache entry is treated as a miss and refetched. Tests cover both paths.

File: cli/lib/update-check.js [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+  }
```
[the remark this answers is not in the document]
Comment (abeltrano) [reply cli/lib/update-check.js:183]: Added in 4b3f564. `checkForUpdate` now dispatches `readCache` / `writeCache` / `fetchLatest` through `_internals` so they're stubbable. New tests cover: fresh cache hit (no fetch), expired cache triggers fetch + rewrite, pkg-name mismatch ignores cache, unparseable cached `latest` is treated as a miss, unparseable fetched `latest` is not cached, and `null` from fetch returns `null`.

**Review by Alan-Jowett — APPROVED** [review]
